src/java/org/apache/cassandra/db/compaction/ShardManagerNodeAware.java

```diff
@@ -102,15 +102,19 @@ private Token[] findTokenAlignedSplitPoints(Token[] sortedTokens, int shardCount
         // Short circuit on equal
         if (sortedTokens.length == shardCount - 1)
             return sortedTokens;
+        // TODO how do we handle 1 shard. Should it be the token min value or the first value in the sorted tokens array?
```

Comment:
A zero-length array should work.

Reply:
I was getting errors when returning a zero length array because `NodeAlignedShardTracker` tries to access the first entry for the start shard. Perhaps I need to update `NodeAlignedShardTracker` to always return the minimum token before iterating over the list of tokens produced by `findTokenAlignedSplitPoints`?

Reply:
I went with the zero length array.